src/main/java/spireMapOverhaul/zones/gremlinTown/events/GremlinWheel.java

```diff
@@ -0,0 +1,387 @@
+import com.badlogic.gdx.math.MathUtils;
+import com.megacrit.cardcrawl.cards.DamageInfo;
+import com.megacrit.cardcrawl.core.CardCrawlGame;
+import com.megacrit.cardcrawl.core.Settings;
+import com.megacrit.cardcrawl.dungeons.AbstractDungeon;
+import com.megacrit.cardcrawl.events.AbstractImageEvent;
+import com.megacrit.cardcrawl.events.GenericEventDialog;
+import com.megacrit.cardcrawl.helpers.Hitbox;
+import com.megacrit.cardcrawl.helpers.ImageMaster;
+import com.megacrit.cardcrawl.helpers.MathHelper;
+import com.megacrit.cardcrawl.helpers.MonsterHelper;
+import com.megacrit.cardcrawl.helpers.controller.CInputActionSet;
+import com.megacrit.cardcrawl.helpers.input.InputHelper;
+import com.megacrit.cardcrawl.localization.EventStrings;
+import com.megacrit.cardcrawl.vfx.RainingGoldEffect;
+import spireMapOverhaul.SpireAnniversary6Mod;
+import spireMapOverhaul.zones.gremlinTown.GremlinTown;
+
+import java.util.ArrayList;
+import java.util.Collections;
+
+import static spireMapOverhaul.SpireAnniversary6Mod.makeID;
+import static spireMapOverhaul.SpireAnniversary6Mod.makeImagePath;
+import static spireMapOverhaul.util.Wiz.*;
+
+public class GremlinWheel extends AbstractImageEvent {
+    public static final String ID = makeID(GremlinWheel.class.getSimpleName());
+    private static final EventStrings eventStrings;
+    private static final String NAME;
+    private static final String[] DESCRIPTIONS;
+    private static final String[] OPTIONS;
+    private static final String IMAGE_PATH;
+    private CUR_SCREEN screen;
+    private boolean startSpinFake;
+    private boolean finishSpinFake;
+    private boolean startSpinReal;
+    private boolean doneSpinning;
+    private boolean bounceIn;
+    private float bounceTimer;
+    private float animTimer;
+    private float spinVelocity;
+    private boolean buttonPressed;
+    private Hitbox buttonHb;
+    private Texture wheelImg;
+    private Texture arrowImg;
+    private Texture buttonImg;
+    private static final float START_Y;
+    private static final float TARGET_Y;
+    private float imgX;
+    private float imgY;
+    private float wheelAngle;
+    private static final int WHEEL_W = 1024;
+    private static final int ARROW_W = 512;
+    private static final float ARROW_OFFSET_X;
+    private Color color;
+    private static final float HP_LOSS = 0.1f;
+    private static final float HP_LOSS_A15 = 0.15F;
+    private static final float RESULT_ANGLE_REAL;
+    private static final float RESULT_ANGLE_FAKE;
+    private static final String WHEEL_PATH = makeImagePath("events/GremlinTown/Wheel.png");
+    private float decisionTimer;
+    private boolean deciding;
+    private ArrayList<Integer> ops;
+    int buttonPanic;
+    int buttonSubmit;
+    int buttonWhat;
+    int buttonFight;
+    private float testTimer;
+
+    private static final int GOLD_BASE = 100;
+    private static final int GOLD_VARIANCE = 25;
+    private static final float DECISION_TIME = 3.0F;
```

Comment:
Could we give the player a bit more time, like 5 seconds instead of 3?

Other than that, while this event breaks some conventions about StS, I think it's fine. You're going to get some amusing feedback from people about this event and this zone as a whole, though 😄.

Reply:
Hmm, maybe the time can be based on ascension level